From fd629b43f9dd44498c329ab069e37ef8e1f8801e Mon Sep 17 00:00:00 2001
From: Christian Berendt <berendt@osism.tech>
Date: Thu, 19 Jan 2023 10:29:16 +0100
Subject: [PATCH] ironic: add ironic_agent_files_directory parameter

With the parameter ironic_agent_files_directory it is possible to provide
the directory for the ironic-agent.kernel and ironic-agent.initramfs
files. By default the parameter is set to the value of node_custom_config.
This corresponds to the existing behaviour.

Change-Id: I53bb0eddc5380713a967356c85897d8df8ce505f
---

diff --git a/ansible/roles/ironic/defaults/main.yml b/ansible/roles/ironic/defaults/main.yml
index 7684cf9..54078c1 100644
--- a/ansible/roles/ironic/defaults/main.yml
+++ b/ansible/roles/ironic/defaults/main.yml
@@ -296,6 +296,7 @@
 ironic_dev_repos_pull: "{{ kolla_dev_repos_pull }}"
 ironic_dev_mode: "{{ kolla_dev_mode }}"
 ironic_source_version: "{{ kolla_source_version }}"
+ironic_agent_files_directory: "{{ node_custom_config }}"
 
 
 ####################
diff --git a/ansible/roles/ironic/tasks/config.yml b/ansible/roles/ironic/tasks/config.yml
index ee9ef5a..c5975ef 100644
--- a/ansible/roles/ironic/tasks/config.yml
+++ b/ansible/roles/ironic/tasks/config.yml
@@ -170,7 +170,7 @@
   vars:
     service: "{{ ironic_services['ironic-tftp'] }}"
   copy:
-    src: "{{ node_custom_config }}/ironic/{{ item }}"
+    src: "{{ ironic_agent_files_directory }}/ironic/{{ item }}"
     dest: "{{ node_config_directory }}/ironic-tftp/{{ item }}"
     mode: "0660"
   become: true
@@ -191,7 +191,7 @@
   vars:
     service: "{{ ironic_services['ironic-http'] }}"
   copy:
-    src: "{{ node_custom_config }}/ironic/{{ item }}"
+    src: "{{ ironic_agent_files_directory }}/ironic/{{ item }}"
     dest: "{{ node_config_directory }}/ironic-http/{{ item }}"
     mode: "0660"
   become: true
diff --git a/ansible/roles/ironic/tasks/precheck.yml b/ansible/roles/ironic/tasks/precheck.yml
index 7150294..6de0731 100644
--- a/ansible/roles/ironic/tasks/precheck.yml
+++ b/ansible/roles/ironic/tasks/precheck.yml
@@ -51,7 +51,7 @@
 
 - name: Checking ironic-agent files exist for Ironic Inspector
   stat:
-    path: "{{ node_custom_config }}/ironic/{{ item }}"
+    path: "{{ ironic_agent_files_directory }}/ironic/{{ item }}"
   delegate_to: localhost
   run_once: True
   register: result
diff --git a/releasenotes/notes/ironic-agent-files-directory-parameter-12868270b1d7f0ea.yaml b/releasenotes/notes/ironic-agent-files-directory-parameter-12868270b1d7f0ea.yaml
new file mode 100644
index 0000000..87748bf
--- /dev/null
+++ b/releasenotes/notes/ironic-agent-files-directory-parameter-12868270b1d7f0ea.yaml
@@ -0,0 +1,7 @@
+---
+features:
+  - |
+    With the parameter ``ironic_agent_files_directory`` it is possible to provide
+    the directory for the ``ironic-agent.kernel`` and ``ironic-agent.initramfs``
+    files. By default the parameter is set to the value of ``node_custom_config``.
+    This corresponds to the existing behaviour.
